src/polar/Pie.tsx

```diff
@@ -345,17 +345,25 @@ const parseCoordinateOfPie = (
     cx?: number | string;
     cy?: number | string;
     innerRadius?: number | string;
-    outerRadius?: number | string;
+    outerRadius?: number | string | ((element: any) => number);
     maxRadius?: number;
   },
   offset: ChartOffset,
+  element: any,
 ): PieCoordinate => {
   const { top, left, width, height } = offset;
   const maxPieRadius = getMaxRadius(width, height);
   const cx = left + getPercentValue(item.cx, width, width / 2);
   const cy = top + getPercentValue(item.cy, height, height / 2);
   const innerRadius = getPercentValue(item.innerRadius, maxPieRadius, 0);
-  const outerRadius = getPercentValue(item.outerRadius, maxPieRadius, maxPieRadius * 0.8);
+
+  let outerRadius;
+  if (typeof item.outerRadius === 'function') {
+    outerRadius = item.outerRadius(element);
+  } else {
+    outerRadius = getPercentValue(item.outerRadius, maxPieRadius, maxPieRadius * 0.8);
+  }
```

Comment:
nit: we could make this a small helper function so we don't have to use a `let`?

Reply:
made the change